packages/Operators/test/tstReferenceHexIntegrationRule.cpp

```diff
@@ -0,0 +1,113 @@
+
+  *: Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+
+  *: Neither the name of the University of Wisconsin - Madison nor the
+  names of its contributors may be used to endorse or promote products
+  derived from this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+  HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   tstReferenceHexIntegrationRule.cpp
+ * \author Stuart Slattery
+ * \date   Wed May 25 12:36:14 2011
+ * \brief  Integration rule function test.
+ */
+//---------------------------------------------------------------------------//
+
+#include "reference_implementation/DTK_ReferenceHexIntegrationRule.hpp"
+#include "reference_implementation/DTK_ReferenceHex.hpp"
+#include "reference_implementation/DTK_ReferenceNode.hpp"
+
+#include <Teuchos_UnitTestHarness.hpp>
+#include <Teuchos_Array.hpp>
+
+//---------------------------------------------------------------------------//
+// Hex test.
+TEUCHOS_UNIT_TEST( ReferenceHexIntegrationRule, hex_test )
+{
+    // Create the nodes;
+    int num_nodes = 8;
+    Teuchos::Array<DataTransferKit::Entity> nodes( num_nodes );
+    nodes[0] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 0, 0.0, 0.0, 0.0 );
+    nodes[1] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 1, 2.0, 0.0, 0.0 );
+    nodes[2] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 2, 2.0, 2.0, 0.0 );
+    nodes[3] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 3, 0.0, 2.0, 0.0 );
+    nodes[4] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 4, 0.0, 0.0, 2.0 );
+    nodes[5] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 5, 2.0, 0.0, 2.0 );
+    nodes[6] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 6, 2.0, 2.0, 2.0 );
+    nodes[7] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 7, 0.0, 2.0, 2.0 );
+    
+    // Make a hex.
+    DataTransferKit::Entity hex = 
+        DataTransferKit::UnitTest::ReferenceHex( 0, 0, nodes );
```

Comment:
This looks pretty damned close to the code in `tstReferenceHex.cpp`. This calls for a `make_hex(...)`.